apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataClientTest.cs

```diff
@@ -120,13 +121,99 @@ public void GetProjectMetadata()
         }
 
         [Fact]
+        public async Task InstanceMetadataChanged()
+        {
+            const string key = "my_instance_key1";
+            const string newValue = "foo";
+
+            var client = MetadataClient.Create();
+            var originalValue = await client.GetCustomInstanceMetadataAsync(key);
+
+            var waitHandle = new ManualResetEventSlim(false);
+            int eventCount = 0;
+            client.InstanceMetadataChanged += (s, e) =>
+            {
+                eventCount++;
+                waitHandle.Set();
+            };
+
+            try
+            {
+                await _fixture.UpdateMetadataAsync($"instance/attributes/{key}", newValue);
+                waitHandle.Wait(TimeSpan.FromSeconds(10));
+                Assert.Equal(1, eventCount);
+                Assert.Equal(newValue, await client.GetCustomInstanceMetadataAsync(key));
+            }
+            finally
```

Comment:
My main concern is if a test expects content at the parent level: if it gets "instance/attributes" or it gets the metadata `Instance` and iterates over all custom keys. Those tests would expect to only find one key. But I guess what I can do it use the guid approach as you say, add a reset mechanism to the emulator, and allow any tests concerned about the parent level do the reset to make sure they start with a clean slate. Does this seem like a better approach?